src/gamescope/wayland/metrics/dbus.rs

```diff
@@ -43,10 +79,26 @@ impl DBusInterface {
     /// Update the metrics from gamescope. This should be called regularly to
     /// get the latest frame metrics. Note that calling this frequently can
     /// interfere with MangoApp framerate stats.
-    pub fn update(&mut self) -> fdo::Result<()> {
-        let msg = self.queue.recv().map_err(MetricsError::from)?;
-        self.state = Some(msg);
-        self.last_update = SystemTime::now();
+    pub async fn update(&mut self) -> fdo::Result<()> {
+        if let Err(err) = self.request_tx.send(()) {
```

Comment:
We should use `send_timeout` here instead to ensure this doesn't block for too long.